lib/galaxy/tools/toolbox/watcher.py

```diff
@@ -151,7 +153,7 @@ def __init__(self, toolbox, observer_class):
         self.start()
 
     def start(self):
-        self.observer.start()
+        register_postfork_function(self.observer.start)
```

Comment:
I'd prefer lib/galaxy/tools/toolbox/watcher.py have nothing to do with uwsgi - in the abstract it really shouldn't and it is meant to be a general library utility useful outside a web server.

In this sense, I'd be much more comfortable with this if it was called:

``lib/galaxy/util/threading.py``

and ``register_postfork_function`` was called something else like ``handle_long_running_function``.